core/src/main/java/com/linecorp/armeria/client/retry/BackoffSpec.java

```diff
@@ -0,0 +1,215 @@
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client.retry;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import com.google.common.base.MoreObjects;
+
+/**
+ * A specification of a {@link Backoff} configuration represented by a string.
+ * The string syntax is a series of comma-separated key-values pairs and each values is a colon-separated.
+ * <ul>
+ *     <li>{@code exponential=[initialDelayMillis:maxDelayMillis:multiplier]}
+ *     is for {@link Backoff#exponential(long, long, double)}
+ *     (maxDelayMillis and multiplier can be omitted)</li>
+ *     <li>{@code jitter=[minJitterRate:maxJitterRate]} is for {@link Backoff#withJitter(double, double)}
+ *     (if one jitter value is specified it will create {@link Backoff#withJitter(double)}</li>
+ *     <li>{@code maxAttempts=[maxAttempts]} is for {@link Backoff#withMaxAttempts(int)}</li>
+ * </ul>
+ * So for example, you can specify it such as {@code exponential=100:10000:2.0,jitter=-0.4:0.3,maxAttempts=100}.
+ * The order of options does not matter, and the {@code specification} needs at least a option.
+ */
+final class BackoffSpec {
+    private static final String SPLIT_OPTIONS = ",";
+    private static final String SPLIT_KEY_VALUES = "=";
+    private static final String SPLIT_VALUES = ":";
+
+    private static final long DEFAULT_INITIAL_DELAY_MILLIS = 200;
+    private static final long DEFAULT_MAX_DELAY_MILLIS = 10000;
+    private static final double DEFAULT_MULTIPLIER = 2.0;
+    private static final double DEFAULT_MIN_JITTER_RATE = -0.2;
+    private static final double DEFAULT_MAX_JITTER_RATE = 0.2;
+    private static final int DEFAULT_MAX_ATTEMPTS = 10;
+
+    private final String specification;
+
+    private boolean exponentialConfigured;
+    private boolean jitterConfigured;
+    private boolean maxAttemptsConfigured;
+
+    long initialDelayMillis = DEFAULT_INITIAL_DELAY_MILLIS;
+    long maxDelayMillis = DEFAULT_MAX_DELAY_MILLIS;
+    double multiplier = DEFAULT_MULTIPLIER;
+    double minJitterRate = DEFAULT_MIN_JITTER_RATE;
+    double maxJitterRate = DEFAULT_MAX_JITTER_RATE;
+    int maxAttempts = DEFAULT_MAX_ATTEMPTS;
+
+    /**
+     * Creates a {@link BackoffSpec} from a string.
+     */
+    static BackoffSpec parse(String specification) {
+        requireNonNull(specification, "specification");
+        final BackoffSpec spec = new BackoffSpec(specification);
+        for (String option : specification.split(SPLIT_OPTIONS)) {
+            spec.parseOption(option.trim());
+        }
```

Comment:
You may want to use `com.google.common.base.Splitter` which simplifies things a lot.